trainers/dataloader.py

```diff
@@ -231,6 +200,28 @@ def _write_tokenized_data(self, tokenized):
                 idx += len(arr_batch)
             arr.flush()
 
+    def get_data(self, split = 'train'):
+        ## load the data
+        if self.loading_shapes[split] is None:
+            data = np.memmap(
+                os.path.join(self.tokenized_data_path, f"{split}.bin"),
+                dtype=np.uint16,
+                mode="r",
+            )
+            self.loading_shapes[split] = (len(data)// self.model_cfg.embedder.byte_context_window, self.model_cfg.embedder.byte_context_window)
+
+            ## reset the data
+            data = None
+
+        ## re-load the data with loading shapes
+        data = np.memmap(
+            os.path.join(self.tokenized_data_path, f"{split}.bin"),
+            dtype=np.uint16,
+            mode="r",
+            shape=self.loading_shapes[split],
```

Comment:
I mean this could just be a subclassed function _get_loading_shape() rather than subclassing full func. but this is good for now lol

Reply:
and tbf this could change more for different ones so happy to keep like this